test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/Configuration/SdkConfigurationTests.cs

```diff
@@ -0,0 +1,133 @@
+// <copyright file="SdkConfigurationTests.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using Xunit;
+
+namespace OpenTelemetry.Configuration.Tests
+{
+    [Collection("xUnitCollectionPreventingTestsThatDependOnSdkConfigurationFromRunningInParallel")]
+    public class SdkConfigurationTests : IDisposable
+    {
+        public SdkConfigurationTests()
+        {
+            ClearEnvVars();
+            SdkConfiguration.Reset();
+        }
+
+        public void Dispose()
+        {
+            ClearEnvVars();
+            SdkConfiguration.Reset();
+        }
+
+        [Fact]
+        public void SdkConfigurationDefaults()
+        {
+            var config = SdkConfiguration.Instance;
+
+            Assert.Null(config.AttributeValueLengthLimit);
+            Assert.Equal(128, config.AttributeCountLimit);
+            Assert.Null(config.SpanAttributeValueLengthLimit);
+            Assert.Equal(128, config.SpanAttributeCountLimit);
+            Assert.Equal(128, config.SpanEventCountLimit);
+            Assert.Equal(128, config.SpanLinkCountLimit);
+            Assert.Equal(128, config.EventAttributeCountLimit);
+            Assert.Equal(128, config.LinkAttributeCountLimit);
```

Comment:
This is a breaking change. I suppose the specification is a bit ambiguous here https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#attribute-limits. An SDK implementing the environment variables is a `SHOULD` per the spec, but it's not clear if the default values are also a `SHOULD`.

We have two options. Make the breaking change and consider this a bug fix. Or maintain the prior behavior and not introduce these new defaults.

Reply:
I'm going to revert this change to defaults for now so we can come back to this conversation. Setting these new defaults mean everyone takes the perf hit described here https://github.com/open-telemetry/opentelemetry-dotnet/pull/3376#discussion_r924971691